pkg/controller/service/with.go

```diff
@@ -353,6 +349,77 @@ func (s *Service) withUserOrServiceAccountAuth(w http.ResponseWriter, r *http.Re
 	return
 }
 
+func (s *Service) withValidatedSsoJWT(w http.ResponseWriter, r *http.Request, f func(ssoJWT models.SsoJWT)) {
+	var ssoRequest models.Auth0SsoRequest
+	if err := read(r, &ssoRequest); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	if s.auth0Audience == "" || s.auth0Domain == nil {
+		http.Error(w, "SSO is not enabled", http.StatusNotImplemented)
+		return
+	}
+	_, claims, err := serviceutils.ParseAndValidateSignedJWT(s.auth0Domain, s.auth0Audience, ssoRequest.IdToken)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	get := func(key string) (string, error) {
+		v, ok := claims[key]
+		if !ok {
+			return "", errors.New("expected JWT claim not found")
+		}
+		value, ok := v.(string)
+		if !ok {
+			return "", errors.New("expected JWT claim to be string")
+		}
+		return value, nil
+	}
+
+	email, err := get("email")
```

Comment:
Couldn't we just make a `Claims` struct with validation?

Reply:
Yeah, that would work too

Reply:
The part I don't like is that we'd have to parse the `json:"sub"` tag and then put it into the `models.SsoJWT` struct. Feels ugly. We might also have to get different claims depending on the provider (which is one of the claims).